lib/services/saml_idp_service_provider_test.go

```diff
@@ -63,6 +67,82 @@ func TestSAMLIdPServiceProviderMarshal(t *testing.T) {
 	require.Equal(t, expected, actual)
 }
 
+func TestFilterSAMLEntityDescriptor(t *testing.T) {
+	tts := []struct {
+		eds           string
+		ok            bool
+		before, after int
+		name          string
+	}{
+		{
+			eds: edBuilder().
+				ACS(saml.HTTPPostBinding, "https://one.example.com/acs").
+				ACS(saml.HTTPPostBinding, "https://two.example.com/acs").
+				Done(),
+			ok:     true,
+			before: 2,
+			after:  2,
+			name:   "no filtering",
+		},
+		{
+			eds: edBuilder().
+				ACS(saml.HTTPPostBinding, "https://example.com/acs").
+				ACS(saml.HTTPPostBinding, "http://example.com/acs").
+				Done(),
+			ok:     true,
+			before: 2,
+			after:  1,
+			name:   "scheme filtering",
+		},
+		{
+			eds: edBuilder().
+				ACS(saml.HTTPArtifactBinding, "https://example.com/acs").
+				ACS(saml.HTTPPostBinding, "https://example.com/acs").
+				Done(),
+			ok:     true,
+			before: 2,
+			after:  1,
+			name:   "binding filtering",
+		},
+		{
+			eds: edBuilder().
+				ACS(saml.HTTPArtifactBinding, "https://example.com/acs").
+				ACS(saml.HTTPPostBinding, "http://example.com/acs").
+				Done(),
+			ok:     false,
+			before: 2,
+			after:  0,
+			name:   "all invalid",
+		},
+	}
+
+	for _, tt := range tts {
+		t.Run(tt.name, func(t *testing.T) {
+			ed, err := samlsp.ParseMetadata([]byte(tt.eds))
+			require.NoError(t, err)
+
+			require.Equal(t, tt.before, getACSCount(ed))
+
+			err = FilterSAMLEntityDescriptor(ed)
+			if !tt.ok {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
```

Comment:
nit: change `tt.ok` from a bool to a `require.ErrorAssertionFunc` and these 5 lines can collapse to a single line.

Reply:
For tests that care about the output of the fallible operation switching to `AssertionFunc` doesn't really save loc, since you have additional checks that are only run for the happy variants (meaning u still need all the same control-flow).  I don't mind switching to `AssertionFunc` if thats something we'd like to standardize on, but IMO for a test like this that isn't worried about the specific error kinds of failures, it's just more visual noise without a meaningful upside.